### DISCUSSION TITLE
Resetting a variable with $queryString gives different behaviour while is defined the same way

### DISCUSSION THREAD

**Discussion by joostvanhoof** [livewire/livewire#3446] [1. Help] [answered]
I have two identical variables in different components. One variable `input` and one `component`.

It's defined like this:

```php
public $input = null;
protected $queryString = ['input'];
protected $rules = [
  'input' => ['int', 'nullable'],
];
$this->reset('input');
```

When I call `$this->reset('input')` the query parameter `?input=` is removed from the URL, which is the desired behaviour.

Now if I do this on `component`:

```php
public $component = null;
protected $queryString = ['component'];
protected $rules = [
  'component' => ['int', 'nullable'],
];
$this->reset('component');
```

And I call `$this->reset('component')` then query parameter becomes `?component=false` which 1) doesn't look nice but more importantly 2) makes the validation fail.

The only thing that I can think of is that the name `component` is problematic?

**Comment by joshhanley** [comment, marked as answer]
@joostvanhoof yeah, using `$component` is the issue, as LW uses that under the hood (I wonder whether that is noted anywhere actually...)

**Reply by joostvanhoof** [reply]
Thanks! Might indeed be a good idea to add that to the Livewire docs.

**Reply by joshhanley** [reply]
@joostvanhoof ok, so I just tested this and it worked fine for me, `$component` reset back to `null.

Also had a look and it doesn't use `$component` under the hood, I was thinking of blade components which do.

Maybe there is something else going on in your component?

**Reply by joostvanhoof** [reply]
I've checked numerous times but must've missed something. When I changed it to `$comp` it worked so going to just leave that for now, there are bigger fish to fry 😃. Thanks!

**Reply by joshhanley** [reply]
Yeah fair enough. No worries 🙂
